packages/sheets-ui/src/services/clipboard/html-to-usm/converter.ts

```diff
@@ -546,3 +439,212 @@ function decodeHTMLEntities(input: string): string {
 
     return input.replace(/&lt;|&gt;|&amp;|&quot;|&#39;|&nbsp;|<br>/g, (match) => entities[match]);
 }
+
+function extractBordersAndKeepOthers(styleString: string, bordersToKeep: string[]) {
+    if (!styleString || !bordersToKeep.length) return '';
+    const borderRegex = /border(-top|-right|-bottom|-left)?\s*:\s*([^;]+);/g;
+
+    const extractedBorders: Record<string, string> = {};
+
+    const withoutBorders = styleString.replace(borderRegex, (match, p1, p2) => {
+        const borderType = `border${p1 || ''}`;
+        extractedBorders[borderType] = p2.trim();
+        return '';
+    });
+
+    let newStyleString = bordersToKeep
+        .map((border) => {
+            if (border === 'border' && extractedBorders[border]) {
+                return `border: ${extractedBorders[border]};`;
+            } else if (extractedBorders[border]) {
+                return `${border}: ${extractedBorders[border]};`;
+            }
+            return '';
+        })
+        .filter((style) => style)
+        .join(' ');
+
+    newStyleString += ` ${withoutBorders.trim()}`;
+
+    return newStyleString.trim();
+}
+
+function parseTableByHtml(htmlElement: HTMLIFrameElement, skeleton?: SpreadsheetSkeleton) {
```

Comment:
This method is too long, and it requires additional comments to improve its understanding.